lib/services/elasticsearch.js

```diff
@@ -152,1421 +167,1360 @@ class ElasticSearch extends Service {
         return this.kuzzle.services.list.internalCache.pexpire(cacheKey, ttl);
       })
       .then(() => this.client.scroll(esRequest))
-      .then(result => formatSearchResults(result))
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => ({
+        scrollId: body.scrollId,
+        hits: body.hits,
+        total: body.total
+      }))
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Search documents from elasticsearch with a query
-   * @param {Request} request
-   * @returns {Promise} resolve documents matching the filter
+   * Searches documents from elasticsearch with a query
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (null), size (null), scroll (null)
+   *
+   * @returns {Promise.<Object>} { scrollId, hits, aggregations, total }
    */
-  search(request) {
-    const esRequest = initESRequest(request, ['from', 'size', 'scroll']);
-
-    esRequest.body = request.input.body;
-
-    if (!request.input.args.includeTrash) {
-      addActiveFilter(esRequest);
-    }
+  search (
+    index,
+    collection,
+    searchBody,
+    { from=null, size=null, scroll=null } = {})
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: searchBody,
+      from,
+      size,
+      scroll
+    };
 
     return this.client.search(esRequest)
-      .then(result => {
-        const formatted = formatSearchResults(result);
-
-        if (formatted.scrollId !== undefined) {
+      .then(({ body }) => {
+        if (body.scrollId) {
           const
             // ms(scroll) may return undefined if in microseconds or in nanoseconds
-            ttl = ms(esRequest.scroll) || ms(this.config.defaults.scrollTTL),
+            ttl = esRequest.scroll && ms(esRequest.scroll)
+              || ms(this.config.defaults.scrollTTL),
             key = scrollCachePrefix + this.kuzzle.constructor.hash(
-              formatted.scrollId
+              body.scrollId
             );
 
           return this.kuzzle.services.list.internalCache.psetex(key, ttl, 0)
-            .then(() => formatted);
+            .then(() => body);
         }
 
-        return formatted;
+        return {
```

Comment:
Here and everywhere else I preferred to re-construct the object returned to keep only relevant informations.  Also even when we keep all properties returned by ES (very rarely), I prefer explicitly re-construct the returned object.